websites.yml

```diff
@@ -14,60 +14,72 @@ content_verification:
 username_patterns:
   behance:
     characters: a-zA-Z0-9_-
-    min_length: 1
+    available_min_length: 4
+    min_length: 3
     max_length: 30
   deviantart:
     characters: a-zA-Z0-9-
+    available_min_length: 5
```

Comment:
@seeeturtle Can you explain the role of ``available_min_length`` here?
Also, why it is in ``websites.yml`` rather than ``test_data.yml`` (in case we need it)

Reply:
@manu-chroma if we use `min_length`, it's too short that almost every username is taken so I add `available_min_length` which means the minimum length that most likely to be available for new users.

Reply:
If `available_min_length` is in `websites.yml`, we can use it to warn or recommend username's length

Reply:
Assuming you came up with ``available_min_length`` by hit-and-trial, I think it would make more sense to move it to ``test_data.yml`` since it is primarily being used for that and I don't think we have any plans to use it in frontend for the time being.

Also, since I wasn't actively following this issue, I'd prefer to have @jayvdb's opinion too

Reply:
actually it's @jayvdb's opinion

Reply:
The idea is that we can use it in the UI as @seeeturtle says "warn or recommend username's length".

While twitter username [`a`](https://twitter.com/a) exists, we know all usernames less than length `available_min_length` (6 for twitter) are taken or invalid.

`min_length` is also based on hit-and-trial, and has mostly been wrong.  They are black boxes, and we've previously used their documentation to determine the `min_length`. For any mildly popular website, any low minimum length their website allows, will quickly become taken.  And their documentation will be outdated.

`min_length` should perhaps be renamed to `min_existent_length`, and known username of this length added to the test data.

Reply:
> min_length should perhaps be renamed to min_existent_length, and known username of this length added to the test data.

``min_length`` needs to be the absolute minimum length which can result in the username that is taken/available on the website i.e. not impossible to have. The point is to not change ``min_length`` irrespective of username not being available for that length. So I think, it is important to have it correct and I think we can really get it right.